### PR TITLE
📘 DOC: Move `site` to a prop instead of directly importing from within the c…

### PR DESCRIPTION
**Pull request by jasikpark**
…omponent.

## Docs

Refactor `<MetaData/>` to be more reusable: `<MetaData {content} {site} canonicalUrl={Astro.request.canonicalUrl}/>`

### CODE REVIEW

**Comment by matthewp** [comment]
👍 looks great

File: docs/src/layouts/Main.astro [hunk 1]
```diff
@@ -23,7 +23,7 @@ if (currentPage) {
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     
     <title>{content.title ? `${content.title} 🚀 ${site.title}` : site.title}</title>
-    <MetaData {content} canonicalURL={Astro.request.canonicalURL}/>
+    <MetaData {content} {site} canonicalURL={Astro.request.canonicalURL}/>
```
Comment (FredKSchott) [inline docs/src/layouts/Main.astro:26]: I missed this in the original MetaData PR: do we want to support `{content}` vs. `content={content}`? I wasn't aware that this worked!

Fine to experiment with this for now in our own site, but whenever we upstream this back to the docs template I don't think if we're ready to make this syntax "blessed" yet.

**Review by FredKSchott — APPROVED** [review]

File: docs/src/layouts/Main.astro (hunk 1 above)
Comment (jasikpark) [reply docs/src/layouts/Main.astro:26]: In the sense of saying "this is the code pattern you should use"? I do like the aspect where it feels similar to the spread API in javascript where you can add a variable to an object without writing something like
```js
const x = {
  "html": html
};
```

Comment (FredKSchott) [reply docs/src/layouts/Main.astro:26]: yea I like that it saves characters, and if it were JSX i would love it, but it feels a bit "not HTML" and we do claim to be a superset of HTML... not too worried about it but just surprised that this works!

**Pull request merged**